skeleton/notification/type/demo.php.twig

```diff
@@ -157,12 +157,14 @@ class demo extends \phpbb\notification\type\base
 	 *
 	 * @param array $data The type specific data
 	 * @param array $pre_create_data Data from pre_create_insert_array()
+{% if REQUIREMENTS.phpbb_version_min matches '/^[\\D]*3\\.1.*$/' %}
 	 *
 	 * @return array Array of data ready to be inserted into the database
+{% endif %}
```

Comment:
Pls remove the `endif` here and fix the tabs below.

See lines 169 etc.

Reply:
```php
	/**
	 * Function for preparing the data for insertion in an SQL query
	 * (The service handles insertion)
	 *
	 * @param array $data The type specific data
	 * @param array $pre_create_data Data from pre_create_insert_array()
{% if REQUIREMENTS.phpbb_version_min matches '/^[\\D]*3\\.1.*$/' %}
	 *
	 * @return array Array of data ready to be inserted into the database
{% endif %}
	 */
	public function create_insert_array($data, $pre_create_data = array())
	{
		$this->set_data('{{ EXTENSION.extension_name|lower }}_sample_name', $data['{{ EXTENSION.extension_name|lower }}_sample_name']);

{% if REQUIREMENTS.phpbb_version_min matches '/^[\\D]*3\\.1.*$/' %}
		return parent::create_insert_array($data, $pre_create_data);
{% else %}
		parent::create_insert_array($data, $pre_create_data);
{% endif %}
	}
```
What needs to be fixed? This results in correct code for both 3.1 and 3.2 minimum versions.

Reply:
![immagine](https://user-images.githubusercontent.com/480857/50184655-fb7d4c00-0315-11e9-83cc-855b4d6d4c8f.png)

Am I wrong the 2 lines as per the red arrows shouldn't be there inside the Docblock?
And the if/else statement below as per the azure box isn't't rightly tabbed.

Reply:
Those files within the `phpbb/skeleton/skeleton/*` are all rendered as template files, not as PHP. So those if statements are rendered properly.

And those if statements are at the beginning of the line, otherwise the tabs/space infront of the if statement might be rendered in the final template file that is rendered.

Reply:
You can see the whole file here

https://github.com/phpbb-extensions/phpbb-ext-skeleton/blob/dfda420590587882da350c28b587f04b194f98c3/skeleton/notification/type/demo.php.twig

Reply:
Should look like
```
	/**
	 * Function for preparing the data for insertion in an SQL query
	 * (The service handles insertion)
	 *
	 * @param array $data The type specific data
	 * @param array $pre_create_data Data from pre_create_insert_array()
	 *
	 * @return array Array of data ready to be inserted into the database
	 */
	public function create_insert_array($data, $pre_create_data = array())
	{
		$this->set_data('{{ EXTENSION.extension_name|lower }}_sample_name', $data['{{ EXTENSION.extension_name|lower }}_sample_name']);

		{% if REQUIREMENTS.phpbb_version_min matches '/^[\\D]*3\\.1.*$/' %}
			return parent::create_insert_array($data, $pre_create_data);
		{% else %}
			parent::create_insert_array($data, $pre_create_data);
		{% endif %}
	}

```

Reply:
Yes, but my point remains: they are TWIG template files, not PHP. They are rendered as display.
See https://github.com/phpbb-extensions/phpbb-ext-skeleton/blob/dfda420590587882da350c28b587f04b194f98c3/helper/packager.php#L114
They are rendered as `assign_display()` and then the content is dumped into a file.

Try the latest commit on this PR out, for both 3.1 and 3.2 min version and you will have a correct nofication type output.

Reply:
No it should not be.
Cause then the result is not correct. Then the `return ..;` or the `parent::` in the final file is not on the same indentation as the `$this->set_data(...`. Aswell as the tabs infron of those indented `{% if ..` statements might be rendered aswell.

And, if the minimum version is 3.2, there should not be a `@return` statement in the DocBlock as it is `void`. Try my last commit here and you will see this is correct.

Reply:
I see, I stand corrected. :+1: